nexus/test-utils/src/lib.rs

```diff
@@ -144,36 +146,53 @@ pub async fn test_setup_with_config<N: NexusServer>(
         .expect("Tests expect to set a port of Clickhouse")
         .set_port(clickhouse.port());
 
-    let server = N::start_and_populate(&config, &logctx.log).await;
-
-    let external_server_addr =
-        server.get_http_server_external_address().await.unwrap();
-    let internal_server_addr = server.get_http_server_internal_address().await;
-
-    let testctx_external = ClientTestContext::new(
-        external_server_addr,
-        logctx.log.new(o!("component" => "external client test context")),
-    );
-    let testctx_internal = ClientTestContext::new(
-        internal_server_addr,
-        logctx.log.new(o!("component" => "internal client test context")),
-    );
+    // Begin starting Nexus.
+    let (nexus_internal, nexus_internal_addr) =
+        N::start_internal(&config, &logctx.log).await;
```

Comment:
I had to split `start_and_populate()` into two steps here.  Previously, this would start Nexus, do rack initialization, and wait for that to finish.  But rack initialization for the tests now expects the configuration for the DNS servers.  That's not known until we start the DNS servers.  That happens when we start the simulated sled agent.  That requires knowing Nexus's internal IP...which we can't know until we start it.  There was a circular dependency here.  So I split this into two steps: one starts Nexus so we can grab its internal API and start the simulated sled agent, and the next step does rack initialization and finishes starting Nexus.